src/lang/base/Syntax.ml

```diff
@@ -558,10 +662,13 @@ module ScillaSyntax (SR : Rep) (ER : Rep) = struct
       | Fun (f, _, body) -> recurser body (f :: bound_vars)
       | TFun (_, body) -> recurser body bound_vars
       | Constr (_, _, es) -> any_is_mem (get_free es bound_vars) blist
-      | App (f, args)
-      | Builtin (f, args) ->
+      | App (f, args) ->
         let args' = f :: args in
         any_is_mem (get_free args' bound_vars) blist
+      | Builtin (_f, args) ->
+        (*let args' = f :: args in -- double check me! *)
+        let args' = args in
```

Comment:
This is correct, although there is now no reason to introduce the `args'` variable.